terraform-provider-kentik-cloudexport/Makefile

```
@@ -0,0 +1,53 @@
+TEST?=$$(go list ./... | grep -v 'vendor')
+HOSTNAME=kentik.com
+NAMESPACE=automation
+NAME=kentik-cloudexport
+BINARY=terraform-provider-${NAME}
+VERSION=0.1.0
+OS_ARCH=linux_amd64
+
+# apiserver address for the provider under test to talk to (for testing purposes)
+APISERVER_ADDR=localhost:9955
+
+
+default: install
+
+build:
+	go build -o ${BINARY}
+
+release:
+	GOOS=darwin GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_darwin_amd64
+	GOOS=freebsd GOARCH=386 go build -o ./bin/${BINARY}_${VERSION}_freebsd_386
+	GOOS=freebsd GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_freebsd_amd64
+	GOOS=freebsd GOARCH=arm go build -o ./bin/${BINARY}_${VERSION}_freebsd_arm
+	GOOS=linux GOARCH=386 go build -o ./bin/${BINARY}_${VERSION}_linux_386
+	GOOS=linux GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_linux_amd64
+	GOOS=linux GOARCH=arm go build -o ./bin/${BINARY}_${VERSION}_linux_arm
+	GOOS=openbsd GOARCH=386 go build -o ./bin/${BINARY}_${VERSION}_openbsd_386
+	GOOS=openbsd GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_openbsd_amd64
+	GOOS=solaris GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_solaris_amd64
+	GOOS=windows GOARCH=386 go build -o ./bin/${BINARY}_${VERSION}_windows_386
+	GOOS=windows GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_windows_amd64
+
+install: build
+	mkdir -p ~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${OS_ARCH}
+	mv ${BINARY} ~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${OS_ARCH}
+
+test:
+	# install dependencies to tests
+	go test -i $(TEST) || exit 1
+
+	# build & run local apiserver
+	go build github.com/kentik/community_sdk_golang/apiv6/localhost_apiserver
+	./localhost_apiserver -addr ${APISERVER_ADDR} -storage internal/provider/CloudExportTestData.json &
+	sleep 1 # let the server some warm up time
+
+	# run tests; set KTAPI_URL to our local apiserver url - otherwise the provider will try to connect to live kentik server
+	echo $(TEST) | KTAPI_URL="http://${APISERVER_ADDR}" xargs go test $(TESTARGS) -run="." -timeout=30s -parallel=4 -count=1 -v || true # "true" to also kill the localhost_apiserver in case when tests fail
```

```suggestion
	KTAPI_URL="http://${APISERVER_ADDR}" go test $(TESTARGS) -timeout=30s ./... || true # "true" to also kill the localhost_apiserver in case when tests fail
```

1. `./...` should be sufficient because it does not match vendor directory since https://go-review.googlesource.com/c/go/+/38745/
2. What is the reason not to keep the default of -parallel parameter (which is GOMAXPROCS)
3. I think that we should enable test caching by default (get rid of `-count=1`)
4. IMO there is no need to use `-v` by default. Developer can pass it if he needs that. But it's subjective, I guess

---

1 - ok
2 - no particular reason, removing it
3 - tests depend on localhost_apiserver and test data; I think caching can lead to problems here
4 - no need indeed, removing

---

Resolved